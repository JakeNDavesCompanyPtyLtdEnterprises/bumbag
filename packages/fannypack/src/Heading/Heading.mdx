---
name: Heading
route: /typography/heading
menu: Typography
---

import { Playground } from 'docz';
import PropsTable from '../_docs/components/PropsTable';
import { Box } from '../primitives/index';
import {Heading} from './Heading';

# Heading

## Import

`import { Heading } from 'fannypack'`

## Basic Usage

By default, `<Heading>` renders a `<h1>` with default margins applied.

<Playground>
  <Heading>Heading 1</Heading>
  <p>Hello world!</p>
</Playground>

> Remember, you can also specify different colors with the `color` prop.

<Playground>
  <Heading color="success">Heading 1</Heading>
  <Heading color="danger">Heading 1</Heading>
</Playground>

## Other headers

Use the `use` prop to change from the default `h1` element to `h2`, `h3`, `h4`, `h5`, `h6`, `span`, or others.

<Playground>
  <Heading>Heading 1</Heading>
  <Heading use="h2">Heading 2</Heading>
  <Heading use="h3">Heading 3</Heading>
  <Heading use="h4">Heading 4</Heading>
  <Heading use="h5">Heading 5</Heading>
  <Heading use="h6">Heading 6</Heading>
</Playground>

## Subheadings

Subheading's are the same size as a normal heading, however have a slightly lower font weight.

<Playground>
  <Heading isSubHeading>Heading 1</Heading>
  <Heading use="h2" isSubHeading>Heading 2</Heading>
  <Heading use="h3" isSubHeading>Heading 3</Heading>
  <Heading use="h4" isSubHeading>Heading 4</Heading>
  <Heading use="h5" isSubHeading>Heading 5</Heading>
  <Heading use="h6" isSubHeading>Heading 6</Heading>
</Playground>

A header can also be easily grouped with a subheading

<Playground>
  <Heading>Jake's Fannypack</Heading>
  <Heading use="h3" isSubHeading>Simply the best</Heading>
  <Heading use="h3">Jake's Fannypack</Heading>
  <Heading use="h5" isSubHeading>Simply the best</Heading>
</Playground>

## Props

<PropsTable component={Heading} />

## Theming

### Schema

```jsx
{
  Heading: {
    base: css | Object,
    h1: css | Object,
    h2: css | Object,
    h3: css | Object,
    h4: css | Object,
    h5: css | Object,
    h6: css | Object,
    subHeading: css | Object
  }
}
```

[Learn more about theming](/styling#theming)
