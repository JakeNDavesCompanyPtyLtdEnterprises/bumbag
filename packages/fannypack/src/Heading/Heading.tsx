import { Box as ReakitBox } from 'reakit';
import classNames from 'classnames';

import { useClassName, createComponent, createElement, createHook } from '../utils';
import { HeadingThemeConfig } from '../types/theme';
import { Box, BoxProps } from '../Box';

import * as styles from './styles';

export type LocalHeadingProps = {
  isSubHeading?: boolean;
};
export type HeadingProps = BoxProps & LocalHeadingProps;

const useProps = createHook<HeadingProps>(
  (props, { themeKey, themeKeyOverride }) => {
    const boxProps = Box.useProps(props);

    const className = useClassName({
      style: styles.Heading,
      styleProps: props,
      themeKey,
      themeKeyOverride,
      prevClassName: boxProps.className,
    });

    return { ...boxProps, className: classNames(className, props.isSubHeading ? 'sub-heading' : 'heading') };
  },
  { defaultProps: { isSubHeading: false }, themeKey: 'Heading' }
);

export const Heading = createComponent<HeadingProps>(
  (props) => {
    const HeadingProps = useProps(props);
    return createElement({ children: props.children, component: ReakitBox, use: props.use, htmlProps: HeadingProps });
  },
  {
    attach: {
      useProps,
      displayName: 'Heading',
    },
    defaultProps: { use: 'h1' },
    themeKey: 'Heading',
  }
);
