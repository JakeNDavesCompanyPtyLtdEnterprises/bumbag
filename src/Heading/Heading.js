// @flow
import React, { type Node } from 'react';
import classNames from 'classnames';

import type { Palette } from '../typesold';
import _Heading from './styled';

type Props = {
  use?: any,
  color?: Palette,
  children: Node,
  className?: string,
  isSubHeading?: boolean
};

const Heading = ({ children, className, isSubHeading, ...props }: Props) => (
  <_Heading
    className={classNames({
      [className || '']: Boolean(className),
      heading: !isSubHeading,
      'sub-heading': isSubHeading
    })}
    isSubHeading={isSubHeading}
    {...props}
  >
    {children}
  </_Heading>
);

Heading.defaultProps = {
  use: undefined,
  className: undefined,
  color: undefined,
  isSubHeading: false
};

export default Heading;
