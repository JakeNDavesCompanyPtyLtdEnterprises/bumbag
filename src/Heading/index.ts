export { default as Heading } from './Heading';
export { default } from './Heading';
