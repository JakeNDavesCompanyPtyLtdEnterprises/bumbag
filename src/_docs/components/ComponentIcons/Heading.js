import React from 'react';

const SvgComponent = props => (
  <svg width={132} height={27} fill="none" {...props}>
    <path
      d="M.69 3.112h5.85v7.124h5.538V3.112h5.876V21h-5.876v-7.02H6.54V21H.69V3.112zm26.86 18.096c-1.439 0-2.722-.286-3.848-.858-1.11-.59-1.976-1.387-2.6-2.392-.624-1.023-.936-2.175-.936-3.458 0-1.317.32-2.487.962-3.51a6.465 6.465 0 0 1 2.6-2.366c1.092-.555 2.305-.832 3.64-.832 1.282 0 2.374.286 3.276.858a5.15 5.15 0 0 1 2.054 2.288c.45.953.676 2.028.676 3.224 0 .312-.018.537-.052.676l-7.8 1.014c.208.607.546 1.049 1.014 1.326.468.26 1.083.39 1.846.39 1.178 0 2.452-.303 3.822-.91l.65 3.484c-1.612.71-3.38 1.066-5.304 1.066zm1.794-8.632c-.278-1.092-.936-1.638-1.976-1.638-.624 0-1.118.234-1.482.702-.364.468-.555 1.075-.572 1.82l4.03-.884zm11.665 8.632a6.088 6.088 0 0 1-3.172-.884 6.881 6.881 0 0 1-2.34-2.444c-.59-1.04-.884-2.193-.884-3.458 0-1.248.295-2.375.884-3.38a6.397 6.397 0 0 1 2.366-2.392 6.18 6.18 0 0 1 3.146-.858c.901 0 1.733.251 2.496.754.78.503 1.37 1.257 1.768 2.262V7.792h5.46V21h-5.46v-2.418c-.867 1.75-2.288 2.626-4.264 2.626zm1.69-4.108c.693 0 1.283-.243 1.768-.728.503-.503.771-1.092.806-1.768v-.208a2.51 2.51 0 0 0-.832-1.768c-.503-.485-1.083-.728-1.742-.728-.728 0-1.352.26-1.872.78a2.48 2.48 0 0 0-.754 1.82c0 .728.26 1.343.78 1.846.52.503 1.135.754 1.846.754zm16.361 4.108a5.839 5.839 0 0 1-3.224-.936 6.744 6.744 0 0 1-2.262-2.47 6.917 6.917 0 0 1-.832-3.302c0-1.23.286-2.357.858-3.38.59-1.04 1.37-1.855 2.34-2.444a6.043 6.043 0 0 1 3.146-.884c1.006 0 1.855.208 2.548.624.711.416 1.283 1.083 1.716 2.002V3.866l5.486-.806V21H63.35v-2.418c-.814 1.75-2.244 2.626-4.29 2.626zm1.716-4.108c.728 0 1.335-.26 1.82-.78.503-.52.754-1.127.754-1.82v-.104c0-.676-.26-1.257-.78-1.742-.52-.503-1.118-.754-1.794-.754-.728 0-1.343.26-1.846.78a2.525 2.525 0 0 0-.728 1.82c0 .728.252 1.343.754 1.846.52.503 1.127.754 1.82.754zm15.295 4.108c-1.456 0-2.591-.468-3.406-1.404-.814-.936-1.222-2.21-1.222-3.822V7.974h5.512v7.774c0 .399.104.71.312.936.226.208.512.312.858.312.347 0 .702-.104 1.066-.312l.286 3.614c-.381.295-.884.52-1.508.676a7.592 7.592 0 0 1-1.898.234zM74.2 6.024c-.814 0-1.508-.251-2.08-.754-.554-.52-.832-1.153-.832-1.898 0-.745.295-1.37.884-1.872A3.035 3.035 0 0 1 74.2.746c.798 0 1.482.26 2.054.78.59.52.884 1.135.884 1.846 0 .763-.294 1.395-.884 1.898a3.07 3.07 0 0 1-2.054.754zm6.792 2.132h5.226v2.652c1.006-2.01 2.6-3.016 4.784-3.016 1.664 0 2.964.598 3.9 1.794.954 1.179 1.43 2.817 1.43 4.914V21h-5.46v-7.046c0-.745-.164-1.317-.494-1.716-.312-.399-.754-.598-1.326-.598-.416 0-.823.139-1.222.416-.398.26-.728.641-.988 1.144-.242.503-.364 1.075-.364 1.716V21h-5.486V8.156zM106.2 26.954c-.953 0-1.863-.095-2.73-.286l.416-3.848c.676.104 1.3.156 1.872.156 1.075 0 1.889-.32 2.444-.962.555-.624.884-1.681.988-3.172-1.023 1.439-2.583 2.158-4.68 2.158a6 6 0 0 1-3.172-.884 6.594 6.594 0 0 1-2.288-2.392 6.722 6.722 0 0 1-.832-3.302c0-1.213.277-2.323.832-3.328a6.336 6.336 0 0 1 2.262-2.418 6 6 0 0 1 3.172-.884c1.057 0 1.976.277 2.756.832a3.697 3.697 0 0 1 1.56 2.184V7.792h5.486v9.906c0 3.033-.693 5.33-2.08 6.89-1.369 1.577-3.371 2.366-6.006 2.366zm0-9.854c.693 0 1.291-.243 1.794-.728.503-.503.771-1.092.806-1.768v-.208c-.035-.659-.312-1.24-.832-1.742-.503-.503-1.092-.754-1.768-.754-.711 0-1.317.26-1.82.78a2.479 2.479 0 0 0-.754 1.82c0 .71.251 1.326.754 1.846a2.48 2.48 0 0 0 1.82.754zm15.388.078h2.184V8.52l-2.184.156v-3.38c.762 0 1.586-.139 2.47-.416.884-.295 1.742-.72 2.574-1.274h2.73v13.572h2.184V21h-9.958v-3.822z"
      fill="#3926A5"
    />
  </svg>
);

export default SvgComponent;
