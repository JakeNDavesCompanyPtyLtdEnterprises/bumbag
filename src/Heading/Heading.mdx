---
name: Heading
route: /typography/heading
menu: Typography
---

import { Playground, PropsTable } from 'docz';
import { Box } from '../primitives/index';
import Heading from './index';

# Heading

## Import

`import { Heading } from 'fannypack'`

## Basic Usage

By default, `<Heading>` renders a `<h1>` with default margins applied.

<Playground>
  <Heading>Heading 1</Heading>
  <p>Hello world!</p>
</Playground>

> Remember, you can also specify different colors with the `color` prop.

<Playground>
  <Heading color="success">Heading 1</Heading>
  <Heading color="danger">Heading 1</Heading>
</Playground>

## Other headers

Use the `use` prop to change from the default `h1` element to `h2`, `h3`, `h4`, `h5`, `h6`, `span`, or others.

<Playground>
  <Heading>Heading 1</Heading>
  <Heading use="h2">Heading 2</Heading>
  <Heading use="h3">Heading 3</Heading>
  <Heading use="h4">Heading 4</Heading>
  <Heading use="h5">Heading 5</Heading>
  <Heading use="h6">Heading 6</Heading>
</Playground>

## Subheadings

Subheading's are the same size as a normal heading, however have a slightly lower font weight.

<Playground>
  <Heading isSubHeading>Heading 1</Heading>
  <Heading use="h2" isSubHeading>Heading 2</Heading>
  <Heading use="h3" isSubHeading>Heading 3</Heading>
  <Heading use="h4" isSubHeading>Heading 4</Heading>
  <Heading use="h5" isSubHeading>Heading 5</Heading>
  <Heading use="h6" isSubHeading>Heading 6</Heading>
</Playground>

A header can also be easily grouped with a subheading

<Playground>
  <Heading>Jake's Fannypack</Heading>
  <Heading use="h3" isSubHeading>Simply the best</Heading>
  <Heading use="h3">Jake's Fannypack</Heading>
  <Heading use="h5" isSubHeading>Simply the best</Heading>
</Playground>

## Props

<PropsTable of={Heading} />

## Theming

### Schema

```jsx
{
  base: string | Object,
  h1: string | Object,
  h2: string | Object,
  h3: string | Object,
  h4: string | Object,
  h5: string | Object,
  h6: string | Object,
  subHeading: string | Object
}
```
