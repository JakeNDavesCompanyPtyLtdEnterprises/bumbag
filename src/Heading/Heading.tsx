import * as React from 'react';
// @ts-ignore
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { HeadingProps as ReakitHeadingProps } from 'reakit/ts/Heading/Heading';

import _Heading from './styled';

export interface LocalHeadingProps {
  children: React.ReactNode;
  className?: string;
  isSubHeading?: boolean;
}
export type HeadingProps = LocalHeadingProps & ReakitHeadingProps;

export const Heading: React.FunctionComponent<LocalHeadingProps> = ({
  children,
  className,
  isSubHeading,
  ...props
}) => (
  <_Heading
    className={classNames({
      [className || '']: Boolean(className),
      heading: !isSubHeading,
      'sub-heading': isSubHeading
    })}
    isSubHeading={isSubHeading}
    {...props}
  >
    {children}
  </_Heading>
);

Heading.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  isSubHeading: PropTypes.bool
};
Heading.defaultProps = {
  className: undefined,
  isSubHeading: false
};

// @ts-ignore
const C: React.FunctionComponent<HeadingProps> = Heading;
export default C;
