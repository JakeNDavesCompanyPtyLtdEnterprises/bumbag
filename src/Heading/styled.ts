// @flow
import { theme } from 'styled-tools';
import { Heading } from 'reakit';

import styled, { css } from '../styled';
// @ts-ignore
import Icon from '../Icon/styled';
import { HeadingProps } from './Heading';

const subHeadingProperties = css`
  font-weight: ${theme('fannypack.fontWeights.semibold')};

  & {
    ${theme('fannypack.Heading.subHeading')};
  }
`;

export default styled(Heading)<HeadingProps>`
  font-weight: ${theme('fannypack.fontWeights.bold')};
  line-height: 1.2;
  margin-bottom: 1rem;

  .heading& + .sub-heading {
    margin-top: -1rem;
  }

  &:last-child {
    margin-bottom: 0;
  }

  h1& {
    font-size: 3em;
    @media (max-width: ${theme('fannypack.layout.mobileBreakpoint')}px) {
      font-size: 2em;
    }
    & {
      ${theme('fannypack.Heading.h1')};
    }
  }
  h2& {
    font-size: 2.5em;
    @media (max-width: ${theme('fannypack.layout.mobileBreakpoint')}px) {
      font-size: 1.8em;
    }
    & {
      ${theme('fannypack.Heading.h2')};
    }
  }
  h3& {
    font-size: 2em;
    @media (max-width: ${theme('fannypack.layout.mobileBreakpoint')}px) {
      font-size: 1.6em;
    }
    & {
      ${theme('fannypack.Heading.h3')};
    }
  }
  h4& {
    font-size: 1.5em;
    @media (max-width: ${theme('fannypack.layout.mobileBreakpoint')}px) {
      font-size: 1.4em;
    }
    & {
      ${theme('fannypack.Heading.h4')};
    }
  }
  h5& {
    font-size: 1.25em;
    @media (max-width: ${theme('fannypack.layout.mobileBreakpoint')}px) {
      font-size: 1.2em;
    }
    & {
      ${theme('fannypack.Heading.h5')};
    }
  }
  h6& {
    font-size: 1em;
    & {
      ${theme('fannypack.Heading.h6')};
    }
  }

  & ${Icon} {
    top: 0.1em;
  }

  & {
    ${props => props.isSubHeading && subHeadingProperties};
  }

  ${theme('fannypack.Heading.base')};
`;
