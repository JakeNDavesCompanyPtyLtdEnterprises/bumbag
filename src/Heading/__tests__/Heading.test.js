import React from 'react';
import render from '../../_utils/tests/render';
import Heading from '../Heading';
import 'jest-styled-components';

it('renders correctly for a basic heading', () => {
  const { container } = render(<Heading>test</Heading>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a heading of a different element', () => {
  const { container } = render(<Heading use="h3">test</Heading>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a subheading', () => {
  const { container } = render(
    <Heading use="h3" isSubHeading>
      test
    </Heading>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a heading with a subheading', () => {
  const { container } = render(
    <div>
      <Heading use="h3">heading</Heading>
      <Heading use="h5" isSubHeading>
        subheading
      </Heading>
    </div>
  );
  expect(container.firstChild).toMatchSnapshot();
});
