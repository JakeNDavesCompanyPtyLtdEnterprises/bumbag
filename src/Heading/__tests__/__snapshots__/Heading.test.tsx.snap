// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for a basic heading 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c1:focus:not(:focus-visible) {
  outline: none;
}

.c0 {
  font-weight: 700;
  line-height: 1.2;
  margin-bottom: 1rem;
}

.heading.c0 + .sub-heading {
  margin-top: -1rem;
}

.c0:last-child {
  margin-bottom: 0;
}

h1.c0 {
  font-size: 3em;
}

h2.c0 {
  font-size: 2.5em;
}

h3.c0 {
  font-size: 2em;
}

h4.c0 {
  font-size: 1.5em;
}

h5.c0 {
  font-size: 1.25em;
}

h6.c0 {
  font-size: 1em;
}

.c0 .c2 {
  top: 0.1em;
}

@media (max-width:480px) {
  h1.c0 {
    font-size: 2em;
  }
}

@media (max-width:480px) {
  h2.c0 {
    font-size: 1.8em;
  }
}

@media (max-width:480px) {
  h3.c0 {
    font-size: 1.6em;
  }
}

@media (max-width:480px) {
  h4.c0 {
    font-size: 1.4em;
  }
}

@media (max-width:480px) {
  h5.c0 {
    font-size: 1.2em;
  }
}

<h1
  class="heading c0 c1"
>
  test
</h1>
`;

exports[`renders correctly for a heading of a different element 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c1:focus:not(:focus-visible) {
  outline: none;
}

.c0 {
  font-weight: 700;
  line-height: 1.2;
  margin-bottom: 1rem;
}

.heading.c0 + .sub-heading {
  margin-top: -1rem;
}

.c0:last-child {
  margin-bottom: 0;
}

h1.c0 {
  font-size: 3em;
}

h2.c0 {
  font-size: 2.5em;
}

h3.c0 {
  font-size: 2em;
}

h4.c0 {
  font-size: 1.5em;
}

h5.c0 {
  font-size: 1.25em;
}

h6.c0 {
  font-size: 1em;
}

.c0 .c2 {
  top: 0.1em;
}

@media (max-width:480px) {
  h1.c0 {
    font-size: 2em;
  }
}

@media (max-width:480px) {
  h2.c0 {
    font-size: 1.8em;
  }
}

@media (max-width:480px) {
  h3.c0 {
    font-size: 1.6em;
  }
}

@media (max-width:480px) {
  h4.c0 {
    font-size: 1.4em;
  }
}

@media (max-width:480px) {
  h5.c0 {
    font-size: 1.2em;
  }
}

<h3
  class="heading c0 c1"
>
  test
</h3>
`;

exports[`renders correctly for a heading with a subheading 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c1:focus:not(:focus-visible) {
  outline: none;
}

.c0 {
  font-weight: 700;
  line-height: 1.2;
  margin-bottom: 1rem;
}

.heading.c0 + .sub-heading {
  margin-top: -1rem;
}

.c0:last-child {
  margin-bottom: 0;
}

h1.c0 {
  font-size: 3em;
}

h2.c0 {
  font-size: 2.5em;
}

h3.c0 {
  font-size: 2em;
}

h4.c0 {
  font-size: 1.5em;
}

h5.c0 {
  font-size: 1.25em;
}

h6.c0 {
  font-size: 1em;
}

.c0 .c3 {
  top: 0.1em;
}

.c2 {
  font-weight: 700;
  line-height: 1.2;
  margin-bottom: 1rem;
}

.heading.c2 + .sub-heading {
  margin-top: -1rem;
}

.c2:last-child {
  margin-bottom: 0;
}

h1.c2 {
  font-size: 3em;
}

h2.c2 {
  font-size: 2.5em;
}

h3.c2 {
  font-size: 2em;
}

h4.c2 {
  font-size: 1.5em;
}

h5.c2 {
  font-size: 1.25em;
}

h6.c2 {
  font-size: 1em;
}

.c2 .c3 {
  top: 0.1em;
}

.c2 {
  font-weight: 600;
}

@media (max-width:480px) {
  h1.c0 {
    font-size: 2em;
  }
}

@media (max-width:480px) {
  h2.c0 {
    font-size: 1.8em;
  }
}

@media (max-width:480px) {
  h3.c0 {
    font-size: 1.6em;
  }
}

@media (max-width:480px) {
  h4.c0 {
    font-size: 1.4em;
  }
}

@media (max-width:480px) {
  h5.c0 {
    font-size: 1.2em;
  }
}

@media (max-width:480px) {
  h1.c2 {
    font-size: 2em;
  }
}

@media (max-width:480px) {
  h2.c2 {
    font-size: 1.8em;
  }
}

@media (max-width:480px) {
  h3.c2 {
    font-size: 1.6em;
  }
}

@media (max-width:480px) {
  h4.c2 {
    font-size: 1.4em;
  }
}

@media (max-width:480px) {
  h5.c2 {
    font-size: 1.2em;
  }
}

<div>
  <h3
    class="heading c0 c1"
  >
    heading
  </h3>
  <h5
    class="sub-heading c2 c1"
  >
    subheading
  </h5>
</div>
`;

exports[`renders correctly for a subheading 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c1:focus:not(:focus-visible) {
  outline: none;
}

.c0 {
  font-weight: 700;
  line-height: 1.2;
  margin-bottom: 1rem;
}

.heading.c0 + .sub-heading {
  margin-top: -1rem;
}

.c0:last-child {
  margin-bottom: 0;
}

h1.c0 {
  font-size: 3em;
}

h2.c0 {
  font-size: 2.5em;
}

h3.c0 {
  font-size: 2em;
}

h4.c0 {
  font-size: 1.5em;
}

h5.c0 {
  font-size: 1.25em;
}

h6.c0 {
  font-size: 1em;
}

.c0 .c2 {
  top: 0.1em;
}

.c0 {
  font-weight: 600;
}

@media (max-width:480px) {
  h1.c0 {
    font-size: 2em;
  }
}

@media (max-width:480px) {
  h2.c0 {
    font-size: 1.8em;
  }
}

@media (max-width:480px) {
  h3.c0 {
    font-size: 1.6em;
  }
}

@media (max-width:480px) {
  h4.c0 {
    font-size: 1.4em;
  }
}

@media (max-width:480px) {
  h5.c0 {
    font-size: 1.2em;
  }
}

<h3
  class="sub-heading c0 c1"
>
  test
</h3>
`;
